File: src/main/java/gg/mineral/server/network/packet/play/clientbound/EntityLookAndRelativeMovePacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.server.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public record EntityLookAndRelativeMovePacket(int entityId, byte deltaX, byte deltaY, byte deltaZ, byte yaw, byte pitch)
        implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        os.writeInt(entityId);
        os.writeByte(deltaX);
        os.writeByte(deltaY);
        os.writeByte(deltaZ);
        os.writeByte(yaw);
        os.writeByte(pitch);
    }

    @Override
    public byte getId() {
        return 0x17;
    }
}

File: server/src/main/kotlin/gg/mineral/server/network/packet/play/clientbound/EntityLookAndRelativeMovePacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class EntityLookAndRelativeMovePacket(
    val entityId: Int, val deltaX: Byte, val deltaY: Byte, val deltaZ: Byte, val yaw: Byte,
    val pitch: Byte
) : Packet.OUTGOING {
    override fun serialize(os: ByteBuf) {
        os.writeInt(entityId)
        os.writeByte(deltaX, deltaY, deltaZ, yaw, pitch)
    }

    override val id: Byte
        get() = 0x17
}
